aws/resource_aws_storagegateway_gateway.go

```diff
@@ -311,6 +337,23 @@ func resourceAwsStorageGatewayGatewayCreate(d *schema.ResourceData, meta interfa
 		}
 	}
 
+	if v, ok := d.GetOk("maintenance_start_time"); ok && len(v.([]interface{})) > 0 {
+		m := v.([]interface{})[0].(map[string]interface{})
+
+		input := &storagegateway.UpdateMaintenanceStartTimeInput{
```

Comment:
lets also add DayOfMonth here while you are at it.

Reply:
done